mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -228,5 +228,14 @@ class TaskExecutorStatus {
         @Nullable
         WorkerId workerId;
         long lastHeartbeatInMs;
+
+        public static TaskExecutorStatus NotFound = new TaskExecutorStatus(
```

Comment:
Why not throw a known Exception instead?

Reply:
I believe that utilizing the same DS for exceptional cases can be problematic. It requires every caller to perform additional checks such as “if (case1) do this else if (case2) do that” and so on. However, if the issue is exceptional in nature and it is automatically surfaced/manifested as such, it becomes more intuitive from the client’s point of view.